src/InlineRelationsTransformer.cpp

```diff
@@ -739,8 +739,9 @@ NullableVector<AstAtom*> getInlinedAtom(AstProgram& program, AstAtom& atom) {
 
             // Create a new atom per new version of the argument
             for (AstArgument* newArgument : argumentVersions.getVector()) {
-                AstAtom* newAtom = atom.clone();
-                newAtom->setArgument(i, std::unique_ptr<AstArgument>(newArgument));
+                auto args = atom.getArguments();
+                args[i] = newArgument;
+                AstAtom* newAtom = new AstAtom(atom, args);
```

Comment:
Avoid repeating the type
```suggestion
                auto* newAtom = new AstAtom(atom, args);
```
Is it safe to use `args` to make a new atom? It looks like they are raw pointers to the contents of unique_ptrs in the old atom.